### ISSUE TITLE
Remove option relation

### ISSUE THREAD

**Issue by bergmorten** [aws-amplify/amplify-js#9520]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Not applicable

### Amplify APIs

DataStore

### Amplify Categories

api

### Environment information

amplify --version
7.6.12

in package.json
"aws-amplify": "^4.3.12"

### Describe the bug

I've the following graphQL model:

```
type Fleet
  @model
{
  id: ID! @primaryKey
  name: String!
  description: String
  vehicles: [Vehicle] @hasMany(indexName: "byFleetVehicle", fields: ["id"])
}


type Vehicle
  @model {
  id: ID! @primaryKey
  name: String!
  fleetId: ID @index(name: "byFleetVehicle", sortKeyFields: ["name"])
  fleet: Fleet @belongsTo(fields: ["fleetId"])
}

```

The Fleet membership of Vehicle is optional, and when I try to remove a Fleet membership from a Vehicle with the following code it does not work. I can se see that the local indexDB storage is correctly altered, but the graphQL call to backend API is incorrect. It does not remove the Fleet, and instead returns the old Vehicle which replace the correct indexDb version.

```
          try {
            const vehicleUpdated = Vehicle.copyOf(vehicle, updated => {
              updated.fleet = undefined;
            });
            await DataStore.save(vehicleUpdated);
          } catch (err) {
            console.error( 'Could not remove vehicle from fleet');
          }
```

This is the GrapQL command sent to backend API:

`{"query":"mutation operation($input: UpdateVehicleInput!, $condition: ModelVehicleConditionInput) {\n  updateVehicle(input: $input, condition: $condition) {\n    id\n    name\n     _version\n    _lastChangedAt\n    _deleted\n    fleet {\n      id\n      _deleted\n    }\n  }\n}\n","variables":{"input":{"id":"c70a37c7-bada-4ed0-9af7-597d1e13c91a","_version":20},"condition":null}}`

As you can see the input is missing fleetId: null 

I think the bug is that fleetId: undefined is not parsed to fleetId: null in the graphQL call.

### Expected behavior

Optional relation should be removed when set to undefined in model and saved.

### Reproduction steps

Make a model with optional relationships. Then create entry with relations in database, and then try to remove relationship with setting value to undefined in javascript/typescript.

```
const vehicleUpdated = Vehicle.copyOf(vehicle, updated => {
              updated.fleet = undefined;
            });
await DataStore.save(vehicleUpdated);
```

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by chrisbonifacio** [comment]
Hi @bergmorten 👋 thank you for raising this issue. It seems you want to remove the reference to a Fleet from a Vehicle, but you also want to keep both records. Is that correct?

**Comment by bergmorten** [comment]
Yes, I want to keep both, and since the relationship is optional (no !) it should be possible. When I store the updated model to datastore with fleet = undefined, I can see that for a flash the relationship is removed in the offline copy. But, when the graphQL  query is sent to mutate the vehicle model it is missing the fleet update (no fleetId: null), and therefor returns the original vehicle model with fleet intact.

**Comment by bergmorten** [comment]
Not very elegant, but I manage to do the update with plain graphQL. The was a trick to get the current version from the Vehicle typescript model since you have not exposed that information.

```typescript
const version = (vehicle as unknown as { _version: number })._version;

const inputVariables: UpdateVehicleMutationVariables = {
    input: {
       id: vehicle.id,
        _version: version,
        fleetId: null,
     },
};
```

**Comment by bobalucard** [comment]
+1 bump to get this resolved. Odd that this functionality doesn't exist in Datastore.

**Comment by mmoulton** [comment]
+1 on letting us configure cascading deletes on a per-relationship basis. This can either take the form of configuration into `DataStore.configure` method, and/or an update to the `@hasOne` / `@hasMany` directives to support a parameter like `cascadeDelete: false`
